LanguageFeatures/Class-modifiers/mixin_restrictions_A04_t04.dart

```diff
@@ -0,0 +1,511 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It's a compile-time error if a class or enum declaration D from
+/// library L has S from library K as a declared mixin, unless:
+/// - S is a mixin or mixin class declaration (necessarily from a post-feature
+///   library), or
+/// - S is a non-mixin class declaration which has Object as superclass and
+///   declares no generative constructor, and either
+///   - K is a pre-feature library, or
+///   - K is a platform library and L is a pre-feature library.
+///
+/// @description Check that it is a compile-time error if a class declaration
+/// mixes in class not declared as `mixin`. Test the case when mixed in classes
+/// defined in the same library
```

Comment:
Oops, we don't have those declarations (such as `Class` or `BaseClass`) in this library.